src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1883,6 +1953,22 @@ public String getDisplayName() {
             return Messages.GitHubSCMSource_DisplayName();
         }
 
+        @Nonnull
+        public Map<String, Object> customInstantiate(@Nonnull Map<String, Object> arguments) {
+            Map<String, Object> arguments2 = new HashMap<>(arguments);
+            arguments2.remove("repositoryUrl");
+            arguments2.remove("isConfiguredByUrlDynamicValue");
+            return arguments2;
+        }
+//
+        @Nonnull
+        public UninstantiatedDescribable customUninstantiate(@Nonnull UninstantiatedDescribable ud) {
+            Map<String, Object> scmArguments = new HashMap<>(ud.getArguments());
```

Comment:
Better to use a `TreeMap` here to preserve the order, since IIRC `ud.getArguments` is also backed by a `TreeMap` or some other kind of sorted map. Might make it so you don't have to change the order of fields in some of the tests.